src/main/java/uk/nhs/digital/nhsconnect/nhais/model/edifact/message/Split.java

```diff
@@ -2,22 +2,22 @@
 
 public class Split {
     public static String[] byColon(String input) {
-        return input.split(":");
+        return splitByDelimiter(input, ":");
     }
 
     public static String[] byPlus(String input) {
-        return input.split("\\+");
+        return splitByDelimiter(input, "\\+");
     }
 
-    /**
-     * Matches an apostrophe NOT preceded by a question mark
-     */
     public static String[] bySegmentTerminator(String input) {
-        return input.split("((?<!\\?)')");
+        return splitByDelimiter(input, "'");
     }
 
-    public static String[] byNewLine(String input) {
-        return input.split("\\n");
+    /**
+     * Matches given delimiter NOT preceded by a odd occurrence of question mark
+     * Works for 1 & 3, Lookbehinds must be fixed-width.
+     */
+    private static String[] splitByDelimiter(String input, String delimiter) {
+        return input.split(String.format("(?<![^\\?]\\?|[^\\?]\\?{3})%s", delimiter));
```

Comment:
regex 'Lookbehinds' has some limitations. I haven't found anything better.
SplitTest.java shows which cases are covered.

Reply:
Why did the much simpler `"((?<!\\?)')"` work?

Reply:
It probably doesn't work for multiple ??? before the delimiter

Reply:
What if... we took input and replaced all `??` by for example `@@` and then applied regex for single `?` and then changed `@@` back to `??` ?
Or just parse those escape characters and then apply regex?